graphql/appointment/mutations.ts

```diff
@@ -51,4 +66,23 @@ export class MutateAppointmentResolver {
         await createGroupAppointments(subcourseId, appointments);
         return true;
     }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    async appointmentUpdate(@Ctx() context: GraphQLContext, @Arg('appointmentToBeUpdated') appointmentToBeUpdated: AppointmentUpdateInput) {
+        const appointment = await getLecture(appointmentToBeUpdated.id);
+        const currentDate = moment();
+        const isPastAppointment = moment(appointment.start).isBefore(currentDate);
+
+        if (isPastAppointment) {
+            throw new Error(`Cannot update past appointment.`);
+        }
+
+        await hasAccess(context, 'Lecture', appointment);
```

Comment:
I'd do the access check right after the appointment query, so that you don't do anything more, if the user doesn't have access.